android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -142,4 +236,449 @@ public void rejectEvent(String eventId) {
+        location.setLongitude(longitude);
+
+        Radar.reverseGeocode(location, new Radar.RadarGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress[] addresses) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (addresses != null) {
+                            map.putArray("addresses", RNRadarUtils.arrayForJson(RadarAddress.toJson(addresses)));
+                        }
+                        promise.resolve(map);
+                    } catch (JSONException e) {
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
+                } else {
+                    promise.reject(status.toString(), status.toString());
+                }
+            }
+        });
+    }
+
+    @ReactMethod
+    public void ipGeocode(final Promise promise) {
+        if (promise == null) {
+            return;
+        }
+
+        Radar.ipGeocode(new Radar.RadarIpGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress address) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (address != null) {
+                            map.putMap("address", RNRadarUtils.mapForJson(address.toJson()));
+                        }
+                        promise.resolve(map);
+                    } catch (JSONException e) {
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
+                } else {
+                    promise.reject(status.toString(), status.toString());
+                }
+            }
+        });
+    }
+
+    @ReactMethod
+    public void getDistance(ReadableMap optionsMap, final Promise promise) {
+        if (promise == null) {
+            return;
+        }
+
+        if (!optionsMap.hasKey("destination")) {
+            promise.reject(Radar.RadarStatus.ERROR_BAD_REQUEST.toString(), Radar.RadarStatus.ERROR_BAD_REQUEST.toString());
+
+            return;
+        }
+
+        ReadableMap originMap = optionsMap.getMap("origin");
+        Location origin = null;
+        if (originMap != null) {
+            double latitude = originMap.getDouble("latitude");
+            double longitude = originMap.getDouble("longitude");
+            origin = new Location("RNRadarModule");
+            origin.setLatitude(latitude);
+            origin.setLongitude(longitude);
+        }
```

Comment:
do we still want to execute the request if `origin == null`? seems like the code will still fall through here..

Reply:
If `origin` is `null`, we call the variant that gets the device's currently location and uses it for `origin`. See https://github.com/radarlabs/react-native-radar/pull/90/files/d3bf9d980e5a8d9ea7923b1659b424025fc6140a#diff-d5f0c15b95a2ea3236e058bc4a2d47b8R678.